src/reducers/intl.js

```diff
@@ -1,5 +1,26 @@
+import {addLocaleData} from 'react-intl';
 import {updateIntl as superUpdateIntl} from 'react-intl-redux';
-import {messages} from '../../locale/messages.json';
+import languages from '../languages.json';
+import messages from '../../locale/messages.json';
 
-export {intlReducer as default} from 'react-intl-redux';
-export const updateIntl = locale => superUpdateIntl(locale, messages[locale] || messages.en);
+export {IntlProvider, intlReducer as default} from 'react-intl-redux';
+
+Object.keys(languages).forEach(locale => {
+    // TODO: will need to handle locales not in the default intl - see www/custom-locales
+    const data = require(`react-intl/locale-data/${locale}`);
+    addLocaleData(data);
+});
+
+// start with English
+export const intlInitialState = {
+    intl: {
+        defaultLocale: 'en',
+        locale: 'en',
+        messages: messages.en
+    }
+};
+
+export const updateIntl = locale => superUpdateIntl({
```

Comment:
FWIW the airbnb javascript style guide (a generally sane and reliable source) explictly recommends against the export-from-import one liner. https://github.com/airbnb/javascript section 10.3: 
> Why? Although the one-liner is concise, having one clear way to import and one clear way to export makes things consistent.